data/scripts/network/debug_assert.lua

```diff
@@ -0,0 +1,16 @@
+local handler = PacketHandler(0xE8)
+
+function handler.onReceive(player, msg)
+	local playerGuid = player:getGuid()
+	if Game.hasDebugAssert(playerGuid) then
```

Comment:
You are missing call to `Game.addDebugAssert` for this to ever have effect.

I see that this is reimplemented from C++, but I think it works differently here. As previously the state of `debugAssertSent` would be reset on reconnection (when character stayed in the game) in protocolgame.h. Here you are keeping that information until the character logs out, but character may stay in the game on crash and reconnect. Am I missing something?

Reply:
yeah! you are right regarding the protocol. I had no other idea than onJoin/Leave